src/wayland/virtual_keyboard/virtual_keyboard_handle.rs

```diff
@@ -192,3 +172,62 @@ where
         }
     }
 }
+
+/// Handle the zwp_virtual_keyboard_v1::keymap request.
+fn update_keymap<D>(data: &VirtualKeyboardUserData<D>, format: u32, fd: OwnedFd, size: usize)
+where
+    D: SeatHandler + 'static,
+{
+    // Only libxkbcommon compatible keymaps are supported.
+    if format != KeymapFormat::XkbV1 as u32 {
```

Comment:
Nice!

Reply:
I'm actually curious why you wrote in your original PR that the format is not an enum. Looking at the protocol it seems like it is an enum and that's just how they're represented. Were you just not aware of that back then?

Reply:
Ah well, yes and no, it is an enum, but it is not explicitly defined as an enum required to use it as such. That said, it did not occur to me that I could write it like this, and this is much more self-explanatory.